Mage/src/main/java/mage/game/combat/Combat.java

```diff
@@ -511,9 +521,9 @@ protected void checkAttackRequirements(Player player, Game game) {
                 player.declareAttacker(creature.getId(), defendersToChooseFrom.iterator().next(), game, false);
                 continue;
             }
-            TargetDefender target = new TargetDefender(defendersToChooseFrom, creature.getId());
+            TargetDefender target = new TargetDefender(defendersToChooseFrom);
             target.setRequired(true);
-            target.setTargetName("planeswalker or player for " + creature.getLogName() + " to attack (must attack effect)");
+            target.setTargetName("permanent or player for " + creature.getLogName() + " to attack (must attack effect)");
             if (player.chooseTarget(Outcome.Damage, target, null, game)) {
```

Comment:
New target `TargetDefender` don't use attackerId, so it lost it in some places. Here is a possible problem -- `player.chooseTarget` calls a `TargetDefender` without source ability param (null), so inner code of `TargetDefender` will not find attackerId and will not process `canBeTargetedBy` and getPossibleTargets correctly.

So possible bugs:
- can attack targets with hexproof or protection
- wrong target highlight (some targets with protection will be visible as selectable)

P.S. It's just a theory, I don't test it. If player and creature protection works fine here, then ok.

Reply:
the attacker id wasn't actually being used before so it should be fine as is, plus it has `notTarget` set  to true so it doesn't matter if something has hexproof or protection anyway